waku/README.md

```diff
@@ -0,0 +1,287 @@
+
+This means that breaking changes will necessitate a new protocol version and a new version sub-package. The packages follow the naming convention of `v*` where `*` represents the major / breaking version number of the protocol.
+
+Currently the package has the following version sub-packages:
+
+- [version 0 - `v0`](./v0)
+- [version 1 - `v1`](./v1)
+
+## What does this package do? 
+
+The basic function of this package is to implement the [waku specifications](https://github.com/vacp2p/specs/blob/master/specs/waku/waku-1.md), and provide the `status-go` binary with the ability to send and receive messages via Waku.
+
+## Waku package files
+
+  - [waku.go](#wakugo)
+  - [api.go](#apigo)
+  - [config.go](#configgo)
+  - [mailserver.go](#mailservergo)
+  - [common](#common)
+    - [bloomfilter.go](#bloomfiltergo)
+    - [const.go](#constgo)
+    - [envelope.go](#envelopego)
+    - [errors.go](#errorsgo)
+    - [events.go](#eventsgo)
+    - [filter.go](#filtergo)
+    - [helpers.go](#helpersgo)
+    - [message.go](#messagego)
+    - [metrics.go](#metricsgo)
+    - [protocol.go](#protocolgo)
+    - [rate_limiter.go](#rate_limitergo)
+    - [topic.go](#topicgo)
+  - [Versioned](#versioned)
+     - [const.go](#version-constgo)
+     - [init.go](#version-initgo)
+     - [message.go](#version-messagego)
+     - [peer.go](#version-peergo)
+     - [status_options.go](#version-status_optionsgo)
+
+## Root
+
+### `waku.go`
+
+[`waku.go`](./waku.go) serves as the main entry point for the package and where the main `Waku{}` struct lives.
+
+---
+
+### `api.go`
+
+[`api.go`](./api.go) is home to the `PublicWakuAPI{}` struct which provides the waku RPC service that can be used publicly without security implications.
+
+`PublicWakuAPI{}` wraps the main `Waku{}`, making the `Waku{}` functionality suitable for external consumption.
+
+#### Consumption
+
+`PublicWakuAPI{}` is wrapped by `eth-node\bridge\geth.gethPublicWakuAPIWrapper{}`, which is initialised via `eth-node\bridge\geth.NewGethPublicWakuAPIWrapper()` and exposed via `gethWakuWrapper.PublicWakuAPI()` and is finally consumed by wider parts of the application.
+
+#### Notes
+
+It is worth noting that each function of `PublicWakuAPI{}` received an unused `context.Context` parameter. This is originally passed in way higher up the food-chain and without significant refactoring is not a simple thing to remove / change. Mobile bindings depend on the ability to pass in a context.
+
+---
+
+### `config.go`
+
+[`config.go`](./config.go) is home to the `Config{}` struct and the declaration of `DefaultConfig`.
+
+`Config{}` is used to initialise the settings of an instantiated `Waku{}`. `waku.New()` creates a new instance of a `Waku{}` and takes a `Config{}` as a parameter, if nil is passed instead of an instance of `Config{}`, `DefaultConfig` is used. 

```

Comment:
What are the possible config values and what are they defined as with regards to defaults?

Reply:
Addressed with https://github.com/status-im/status-go/pull/1949/commits/39bc04b1958a3bc0b7161613608ee572bab8aa8a